### ISSUE TITLE
Cross-Schema Foreign Key not working with MSSQL

### ISSUE THREAD

**Issue by kjvalencik** [schemaspy/schemaspy#35]
```
Connecting relationships..........16:58:37.313 WARNING: TableService.addForeignKey - Couldn't add FK 'FK_OTHER_fk_name' to table 'org.schemaspy.service.TableService@2784145d' - Column 'col' doesn't exist in table 'Table_In_dbo'
com.microsoft.sqlserver.jdbc.SQLServerException: The result set is closed.
	at com.microsoft.sqlserver.jdbc.SQLServerException.makeFromDriverError(SQLServerException.java:190)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.checkClosed(SQLServerResultSet.java:372)
	at com.microsoft.sqlserver.jdbc.SQLServerResultSet.next(SQLServerResultSet.java:994)
	at org.schemaspy.service.TableService.connectForeignKeys(TableService.java:209)
	at org.schemaspy.service.DatabaseService.connectTables(DatabaseService.java:224)
	at org.schemaspy.service.DatabaseService.gatheringSchemaDetails(DatabaseService.java:64)
	at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:231)
	at org.schemaspy.SchemaAnalyzer.analyzeMultipleSchemas(SchemaAnalyzer.java:129)
	at org.schemaspy.SchemaAnalyzer.analyze(SchemaAnalyzer.java:82)
	at org.schemaspy.Main.main(Main.java:52)
```

**Comment by rafalkasa** [comment]
@kjvalencik thank you for create the issue. I'm start investigating what is the reason of this exception...
This will be fixed as soon as possible!

**Comment by kjvalencik** [comment]
@rafalkasa Thanks for getting back so quickly. I put together a minimal repro script. Let me know if there is anything else I can provide.

SQL to generate a cross schema FK.

```sql
CREATE SCHEMA a
GO
CREATE SCHEMA b
GO

CREATE TABLE a.tbl_a (c int, PRIMARY KEY (c))
GO
CREATE TABLE b.tbl_b (c int, PRIMARY KEY (c), FOREIGN KEY (c) REFERENCES a.tbl_a(c))
GO
```

Full script to reproduce.

```bash
#!/bin/bash

set -euo pipefail

name="mssql-schemaspy"
password="Passw0rd"

function clean_up {
	docker rm -f "$name"  > /dev/null 2>&1 || true
}

trap clean_up EXIT

function start_sql {
docker run -d \
		--name="$name" \
		-e ACCEPT_EULA="Y" \
		-e SA_PASSWORD="$password" \
	microsoft/mssql-server-linux:latest \
		> /dev/null
}

function exec_sql {
	docker run -i --rm \
			--net="container:${name}" \
			--entrypoint="sqlcmd" \
		tsgkadot/mssql-tools:latest \
			-d "${1-test}" -S 127.0.0.1 -U sa -P "$password"
}

function wait_for_sql {
	until is_ready > /dev/null; do
		sleep 1
	done
}

function is_ready {
	exec_sql master << EOF
SELECT state
FROM master.sys.databases
GO
EOF
}

function create_db {
	exec_sql master << EOF
CREATE DATABASE test
GO
EOF
}

function create_schemas {
	exec_sql << EOF
CREATE SCHEMA a
GO
CREATE SCHEMA b
GO
EOF
}

function create_tables {
	exec_sql << EOF
CREATE TABLE a.tbl_a (c int, PRIMARY KEY (c))
GO
CREATE TABLE b.tbl_b (c int, PRIMARY KEY (c), FOREIGN KEY (c) REFERENCES a.tbl_a(c))
GO
EOF
}

function schema_spy {
	docker run --rm --net="container:${name}" -v $(pwd):/build \
	    schemaspy:latest \
	    -t mssql05 \
	    -dp /opt/schemaspy/lib/sqljdbc4.jar \
	    -db test \
	    -host localhost \
	    -port 1433 \
	    -all \
	    -u sa \
	    -p "$password" \
	    -o /build
}

start_sql
wait_for_sql
create_db
create_schemas
create_tables
schema_spy
```

**Comment by rafalkasa** [comment]
@kjvalencik thank you very much for script that help reproduce the issue. This is more than appreciated.

**Comment by rafalkasa** [comment]
@kjvalencik problem with Cross-Schema Foreign Keys was fixed and will be ready in version 6.0.0-rc1 to this moment in time you need to compile SchemeSpy yourself.

**Comment by kjvalencik** [comment]
Confirmed fixed. Thanks! :+1:
